### DISCUSSION TITLE
How can i share kyuubi session by jdbc ?

### DISCUSSION THREAD

**Discussion by dik111** [apache/kyuubi#3036] [Q&A] [answered]
Hi Teams:
  For some reason, I need session sharing based on JDBC, such as :
```
SessionA with JDBC A: jdbc:hive2://127.0.0.1:10009/default;
SessionB with JDBC B: jdbc:hive2://127.0.0.1:10009/default;#spark.num.executors=5;spark.executor.memory=10G
SessionC with JDBC C: jdbc:hive2://127.0.0.1:10009/default;#spark.num.executors=10;spark.executor.memory=20G
...
```
I tried to find out if there is a configuration item that can meet this requirement by reading the documentation, but I couldn't find it, Is there any configuration item I missed ? 
（`kyuubi.engine.share.level CONNECTION` doesn't meet my needs ）

**Comment by yaooqinn** [comment]
Not quite understand your question

**Comment by turboFei** [comment, marked as answer]
```
SessionA with JDBC A: jdbc:hive2://127.0.0.1:10009/default#kyuubi.engine.pool.name=session_a;kyuubi.engine.pool.size=1
SessionB with JDBC B: jdbc:hive2://127.0.0.1:10009/default;#spark.num.executors=5;spark.executor.memory=10G;kyuubi.engine.pool.name=session_b;kyuubi.engine.pool.size=1
SessionC with JDBC C: jdbc:hive2://127.0.0.1:10009/default;#spark.num.executors=10;spark.executor.memory=20G;kyuubi.engine.pool.name=session_c;kyuubi.engine.pool.size=1
```

Is this the answer?

**Reply by dik111** [reply]
Yes, I hope to be able to distinguish  sessions through jdbc, because in our platform, users can adjust kyuubi-spark parameters by themselves, for example: spark.num.executors. I want the same parameters to be in a kyuubi session so that the user doesn't need to wait for session to start again.

**Reply by turboFei** [reply]
FYI: `kyuubi.engine.pool.name=label;kyuubi.engine.pool.size=1`

**Reply by dik111** [reply]
You are right, but I found that setting kyuubi.engine.share.level.subdomain seems to be able to meet my needs, can you tell me more about this kyuubi.engine.pool.name? I read the documentation and it seems a bit too simplistic.
